### PR TITLE
[Perf] Improve Fish Speech S2 Pro inference performance

### PR DESCRIPTION
**Pull request by Sy0307**
## Purpose

This PR focuses on Fish Speech S2 Pro inference performance based on #1798 . **Waiting for #1798 merge.**

The current diff is shown against `main`, so it also contains the full Fish Speech S2 Pro integration stack. However, the main intent of this PR is the performance work on top of that support.

The main optimizations are:

- batch DAC decode in Stage-1 instead of decoding each request separately
- keep `last_slow_ar_hidden` GPU-resident across decode steps to avoid the per-step GPU -> CPU -> GPU round-trip
- replace connector busy-spin polling with `threading.Condition()` wakeups plus bounded backoff
- store async chunk frames as tensors and pack them with tensor ops instead of repeated `.cpu().tolist()` / `torch.tensor(list)` reconstruction
- keep the guarded Fast AR compiled path for the single-request fast path

In practice, these changes reduce:
- repeated small DAC decode launches under concurrency
- per-step host/device copies in the Slow AR loop
- connector-side CPU polling overhead
- Python-heavy chunk packing overhead in the streaming path

This improves both:
- single-request latency / RTF
- loaded throughput under concurrency

## Test Result

Measured on RTX 5090 with `fishaudio/s2-pro`.

The comparison below uses the same config before and after the perf changes:
- `stage0/stage1 max_batch_size=8`
- `max_inflight=8`

### Single request

| Metric | Before perf changes | After perf changes | Delta |
|---|---:|---:|---:|
| RTF | 0.446 | 0.392 | -12.0% |
| TTFP | 0.121s | 0.114s | -5.8% |
| Request throughput | 0.381 req/s | 0.432 req/s | +13.6% |

### Concurrency = 8

| Metric | Before perf changes | After perf changes | Delta |
|---|---:|---:|---:|
| RTF | 1.113 | 0.760 | -31.7% |
| TTFP | 0.342s | 0.287s | -16.2% |
| Request throughput | 1.204 req/s | 1.757 req/s | +46.0% |
| Audio throughput | 7.099 s/s | 10.363 s/s | +46.0% |

### Concurrency = 16

| Metric | Before perf changes | After perf changes | Delta |
|---|---:|---:|---:|
| RTF | 1.629 | 1.159 | -28.9% |
| TTFP | 3.547s | 2.567s | -27.6% |
| Request throughput | 1.251 req/s | 1.754 req/s | +40.2% |
| Audio throughput | 7.376 s/s | 10.343 s/s | +40.2% |

Validation:
- transfer adapter unit tests passed
- Fish Speech end-to-end benchmark completed for `c=1`, `c=8`, and `c=16`

cc @linyueqian

### CODE REVIEW

**Comment by linyueqian** [comment]
@hsliuustc0106 i will merge #1798 first

**Comment by hsliuustc0106** [comment]
@vllm-omni-reviewer

**Comment by hsliuustc0106** [comment]
resolve conflicts please

**Comment by Sy0307** [comment]
> resolve conflicts please

Resolved.

**Comment by Sy0307** [comment]
PTAK @linyueqian @hsliuustc0106

**Comment by Sy0307** [comment]
> If all requests in a batch have invalid/empty codes, `valid_codes_qf` is empty and `valid_codes_qf[0].device` (line ~228) raises `IndexError`. Please add an early return guard:
> 
> ```python
> if not valid_codes_qf:
>     return audios, srs
> ```
> 
> before the `feature_lengths = torch.tensor(...)` block.

```python
        if not valid_codes_qf:
            return OmniOutput(
                text_hidden_states=None,
                multimodal_outputs={
                    "model_outputs": [empty] * num_req,
                    "sr": [sr_tensor] * num_req,
                },
            )
```

We have set some guards in code. This is enough?

**Comment by linyueqian** [comment]
> > If all requests in a batch have invalid/empty codes, `valid_codes_qf` is empty and `valid_codes_qf[0].device` (line ~228) raises `IndexError`. Please add an early return guard:
> > ```python
> > if not valid_codes_qf:
> >     return audios, srs
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > before the `feature_lengths = torch.tensor(...)` block.
> 
> ```python
>         if not valid_codes_qf:
>             return OmniOutput(
>                 text_hidden_states=None,
>                 multimodal_outputs={
>                     "model_outputs": [empty] * num_req,
>                     "sr": [sr_tensor] * num_req,
>                 },
>             )
> ```
> 
> We have set some guards in code. This is enough?

i think so

**Comment by linyueqian** [comment]
resolve conflicts and should be good to go.

**Comment by linyueqian** [comment]
fix pre-commit please

**Comment by Sy0307** [comment]
> fix pre-commit please

Fixed.
